src/Otel/Internal/Client.hs

```diff
@@ -82,26 +86,25 @@ data OtelClient = OtelClient
 startOtelClient :: ResourceAttributes -> OtelClientParameters -> IO OtelClient
 startOtelClient resourceAttributes' OtelClientParameters {..} = do
   httpManager <- newManager defaultManagerSettings
-  parsedUrlLogRequest <- parseRequest logEndpoint
-  let logRequestBase =
-        parsedUrlLogRequest
-          { method = "POST"
-          , requestHeaders = [("Content-Type", "application/x-protobuf")]
-          }
-  parsedUrlTraceRequest <- parseRequest traceEndpoint
-  let traceRequestBase =
-        parsedUrlTraceRequest
-          { method = "POST"
-          , requestHeaders = [("Content-Type", "application/x-protobuf")]
-          }
+  logRequestBase <- setProtobufRequest <$> parseRequest logEndpoint
+  traceRequestBase <- setProtobufRequest <$> parseRequest traceEndpoint
   let resourceAttributes = toOtelAttributes resourceAttributes'
   logQueueSet <- newOtelQueueSetIO queueSize
   traceQueueSet <- newOtelQueueSetIO queueSize
   let client = OtelClient {..}
-  void . forkIO $ runLogClientProcess client
-  void . forkIO $ runTraceClientProcess client
+  when (not logEnabled && not traceEnabled) $
+    putStrLn "WARNING: Logging and Tracing are disabled."
+  when logEnabled . void . forkIO $ runLogClientProcess client
```

Comment:
If I recall correctly, this is not enough. You need to disable the insertion into the queues as well, or else the app will hang when the queues will fill.

Reply:
I forgot to patch `runOtelStatic` and the `log` function. I have to admit that I am not sure the current code is good enough. I'm open to hearing proposals :)